addons/l10n_sa_edi/models/account_edi_format.py

```diff
@@ -418,7 +418,7 @@ def _set_missing_partner_fields(missing_fields, name):
             errors.append(_set_missing_partner_fields(supplier_missing_info, _("Supplier")))
         if customer_missing_info:
             errors.append(_set_missing_partner_fields(customer_missing_info, _("Customer")))
-        if invoice.invoice_date > date.today():
+        if invoice.invoice_date > fields.Date.context_today(self):
```

Comment:
You should also remove the import

Reply:
Thinking the same, but it would already solve the problem. @mebe-odoo

Reply:
Still need to remove the import @jco-odoo 

Also, forcing the locale, won't it make issue in the test suite because of discrepancies between the test user's locale and the hardcoded locale)

Reply:
@william-andre I see that in the tests, there is freeze_time and the invoice date is set on purpose.  Weirdly we put the timezone at -3 instead of +3 but we do not do later than 3pm, so it should be fine.